### PR TITLE
Fix GH-1656: Check whether a URL has querystring, based on that add timestamp at the end of URL

### PR DESCRIPTION
**Pull request by SID177**
Check whether a URL has querystring, based on that add timestamp at the end of URL

Fixes https://github.com/rtCamp/rtMedia/issues/1656

### CODE REVIEW

**Comment by SID177** [comment]
Yes it'll work. Any URL which has query parameter will work now. I tested it.

**Comment by SID177** [comment]
Earlier the URL was becoming like this
```
https://MyS3BucketName.s3.eu-west-3.amazonaws.com/localhost/2020/06/22155525/3f083c89-a5c7-492c-938d-467d2d666dc5-1080x675.jpg?X-Amz-Content-Sha256=UNSIGNED-PAYLOAD&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AZIA46ZC73DSWOHCZ53N%2F20200622%2Feu-west-3%2Fs3%2Faws4_request&X-Amz-Date=20200622T145545Z&X-Amz-SignedHeaders=host&X-Amz-Expires=900&X-Amz-Signature=fd92ec9003d6c72983145132cbab7df7506dec923111b2a482e045fb724140773?1592838041 
```

**?1592838041** is added. This break the URL.

The problem here is that, in `Signature` parameter, the expected value is `fd92ec9003d6c72983145132cbab7df7506dec923111b2a482e045fb724140773` but due to our code, it becomes `fd92ec9003d6c72983145132cbab7df7506dec923111b2a482e045fb724140773?1592838041`

But now it becomes like this
```
https://MyS3BucketName.s3.eu-west-3.amazonaws.com/localhost/2020/06/22155525/3f083c89-a5c7-492c-938d-467d2d666dc5-1080x675.jpg?X-Amz-Content-Sha256=UNSIGNED-PAYLOAD&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AZIA46ZC73DSWOHCZ53N%2F20200622%2Feu-west-3%2Fs3%2Faws4_request&X-Amz-Date=20200622T145545Z&X-Amz-SignedHeaders=host&X-Amz-Expires=900&X-Amz-Signature=fd92ec9003d6c72983145132cbab7df7506dec923111b2a482e045fb724140773&1592838041
```

Here the value of `Signature` will remain `fd92ec9003d6c72983145132cbab7df7506dec923111b2a482e045fb724140773`

@umeshnevase

File: app/main/controllers/template/rtmedia-functions.php [hunk 1]
```diff
@@ -578,18 +578,26 @@ function rtmedia_media( $size_flag = true, $echo = true, $media_size = 'rt_media
...
 
-			/**
-			 * Used `set_url_scheme` because `esc_url` breaks the image if there is special characters are there into image name.
-			 * Added by checking the code from "wp-admin/includes/media.php:2740".
-			 * Because in media library, it was not breaking.
-			 *
-			 * Add timestamp to resolve conflict with cache image.
-			 */
-			$html = "<img src='" . set_url_scheme( $src[0] . '?' . time() ) . "' alt='" . esc_attr( $rtmedia_media->post_name ) . "' />";
+			if ( ! empty( $src[0] ) ) {
```
Comment (umeshnevase) [inline app/main/controllers/template/rtmedia-functions.php:583]: @SID177 Please check ! empty( $src) else it will notice that array index

Comment (SID177) [reply app/main/controllers/template/rtmedia-functions.php:583]: @umeshnevase It won't throw any notice, `empty` will handle checking `$src` as well as `$src[0]`.
You can try
```
$a = null;
echo !empty($a[0])?'yes':'no';
```
This code in online PHP interpreter and it'll output `no`.

**Comment by ExoGeek** [comment]
Hi,

Just to let you know that the suggested correction won't work for signed URL (on S3).

As soon as you append a string to the URL, the URL is not valid.

I've just tested with a valid S3 Signed URL. I've append "&1592838041" at the end of the URL and got the error below : 

```
<Code>SignatureDoesNotMatch</Code>
<Message>
The request signature we calculated does not match the signature you provided. Check your key and signing method.
</Message>
```

I suppose the best solution could be to not touch the URL if there's already a querystring.

Regards

**Comment by SID177** [comment]
Hi there,

We've removed the timestamp from URL if its from `amazonaws.com`

**Pull request merged**